jax/_src/numpy/linalg.py

```diff
@@ -531,10 +531,9 @@ def norm(x: ArrayLike, ord: Union[int, str, None] = None,
     raise ValueError(
         f"Invalid axis values ({axis}) for jnp.linalg.norm.")
 
-
 @_wraps(np.linalg.qr)
 @partial(jit, static_argnames=('mode',))
-def qr(a: ArrayLike, mode: str = "reduced") -> Tuple[Array, Array]:
+def qr(a: ArrayLike, mode: str = "reduced") -> Union[Array, Tuple[Array, Array]]:
```

Comment:
Can we use a Union of Literals here?

Reply:
I think we should avoid that; see response at https://github.com/google/jax/pull/12674#discussion_r989313739